tests/test_electricity_demand.py

```diff
@@ -8,23 +8,9 @@
 
 """
 
-import numpy as np
+import matplotlib
 
-from demandlib.examples import power_demand_example
 
 def test_power_example():
-    """Test the results of the power example."""
-
-    ann_el_demands_per_sector = {'g0': 3000,
-                                 'h0': 3000,
-                                 'i0': 3000,
-                                 'i1': 5000,
-                                 'i2': 6000,
-                                 'g6': 5000}
-
-    demands = power_demand_example.power_example(
-        ann_el_demand_per_sector=ann_el_demands_per_sector,
-        testmode=True).sum()/4
-
-    for key in ann_el_demands_per_sector:
-        assert np.isclose(demands[key], ann_el_demands_per_sector[key])
+    matplotlib.use("Agg")
+    import demandlib_examples.power_demand_example  # noqa: F401
```

Comment:
The `import` might incidentally do the correct thing, but `exec(open("filename.py").read())` should be the clean way. Alternatively, the example could have a function that can be imported and called from her. (And `if __name__ == "__main__"`: in the example script file.)

Reply:
exec(...) will change the way __file__ is interpreted. Thus, the test script for the examples has to be in the same location the examples are. (Alternatively, only function in the test scrips are tested but not the full scripts.)